homeassistant/components/bring/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Bring! integration."""
+from __future__ import annotations
+
+import logging
+
+from python_bring_api.bring import Bring
+from python_bring_api.exceptions import (
+    BringAuthException,
+    BringParseException,
+    BringRequestException,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import BringDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.TODO]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Bring! from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    email = entry.data[CONF_EMAIL]
+    password = entry.data[CONF_PASSWORD]
+
+    bring = Bring(email, password)
+
+    try:
+        await hass.async_add_executor_job(bring.login)
+        await hass.async_add_executor_job(bring.loadLists)
```

Comment:
We don't want to switch contexts a lot, can we maybe make an inline function we run in the executor?

Reply:
You mean group both together?

Reply:
Yes:
```py
def get_data() -> None:
    bring.login()
    bring.loadLists()

await hass.async_add_executor_job(get_data)
```